[zh] Translate docs/reference/labels-annotations-taints to Chinese (#32255)
kubernetes/website: content/zh/docs/reference/labels-annotations-taints/_index.md

If a Node is in a cloud provider specified shutdown state, the Node gets tainted accordingly with `node.cloudprovider.kubernetes.io/shutdown` and the taint effect of `NoSchedule`.
-->
### node.cloudprovider.kubernetes.io/shutdown {#node-cloudprovider-kubernetes-io-shutdown}

例子：`node.cloudprovider.kubernetes.io/shutdown:NoSchedule`

如果 Node 处于云提供商指定的关闭状态，则 Node 会相应地被 `node.cloudprovider.kubernetes.io/shutdown` 和 `NoSchedule` 的污染效果污染。

<!--
### pod-security.kubernetes.io/enforce

Example: `pod-security.kubernetes.io/enforce: baseline`

Used on: Namespace

Value **must** be one of `privileged`, `baseline`, or `restricted` which correspond to
[Pod Security Standard](/docs/concepts/security/pod-security-standards) levels. Specifically,
the `enforce` label _prohibits_ the creation of any Pod in the labeled Namespace which does not meet
the requirements outlined in the indicated level.

See [Enforcing Pod Security at the Namespace Level](/docs/concepts/security/pod-security-admission)
for more information.
-->
### pod-security.kubernetes.io/enforce {#pod-security-kubernetes-io-enforce}

例子：`pod-security.kubernetes.io/enforce: baseline`

用于：Namespace

> 值**必须**是 `privileged` 、 `baseline` 或 `restricted` 之一，它们对应于

Reviewer: ```suggestion
值**必须**是 `privileged`、`baseline` 或 `restricted` 之一，它们对应于
```

> [ Pod 安全标准](/zh/docs/concepts/security/pod-security-standards) 级别。具体来说，`enforce` 标签 _禁止_ 在带标签的 Namespace 中创建任何不符合指示级别概述要求的 Pod。

Reviewer: ```suggestion
[ Pod 安全标准](/zh/docs/concepts/security/pod-security-standards) 级别。
特别地，`enforce` 标签 **禁止** 在带标签的 Namespace 中创建任何不符合指示级别要求的 Pod。
```

> 请参阅 [在 Namespace 级别实施 Pod 安全性](/zh/docs/concepts/security/pod-security-admission)了解更多信息。

Reviewer: ```suggestion
请参阅[在名字空间级别实施 Pod 安全性](/zh/docs/concepts/security/pod-security-admission)了解更多信息。
```

<!--
### pod-security.kubernetes.io/enforce-version

Example: `pod-security.kubernetes.io/enforce-version: {{< skew latestVersion >}}`

Used on: Namespace

Value **must** be `latest` or a valid Kubernetes version in the format `v<MAJOR>.<MINOR>`.
This determines the version of the [Pod Security Standard](/docs/concepts/security/pod-security-standards) 
policies to apply when validating a submitted Pod.

See [Enforcing Pod Security at the Namespace Level](/docs/concepts/security/pod-security-admission)
for more information.
-->
### pod-security.kubernetes.io/enforce-version {#pod-security-kubernetes-io-enforce-version}

例子：`pod-security.kubernetes.io/enforce-version: {{< skew latestVersion >}}`

用于：Namespace

值**必须**是 `latest` 或格式为 `v<MAJOR>.<MINOR>` 的有效 Kubernetes 版本。

> 这决定了在验证提交的 Pod 时要应用的 [Pod 安全标准](/zh/docs/concepts/security/pod-security-standards) 策略的版本。

Reviewer: ```suggestion
此注解决定了在验证提交的 Pod 时要应用的 [Pod 安全标准](/zh/docs/concepts/security/pod-security-standards)策略的版本。
```

> 请参阅 [在命名空间级别实施 Pod 安全性](/zh/docs/concepts/security/pod-security-admission)了解更多信息。

Reviewer: ```suggestion
请参阅[在名字空间级别实施 Pod 安全性](/zh/docs/concepts/security/pod-security-admission)了解更多信息。
```

<!--
### pod-security.kubernetes.io/audit

Example: `pod-security.kubernetes.io/audit: baseline`

Used on: Namespace

Value **must** be one of `privileged`, `baseline`, or `restricted` which correspond to
[Pod Security Standard](/docs/concepts/security/pod-security-standards) levels. Specifically,
the `audit` label does not prevent the creation of a Pod in the labeled Namespace which does not meet
the requirements outlined in the indicated level, but adds an audit annotation to that Pod.

See [Enforcing Pod Security at the Namespace Level](/docs/concepts/security/pod-security-admission)
for more information.
-->
### pod-security.kubernetes.io/audit {#pod-security-kubernetes-io-audit}

例子：`pod-security.kubernetes.io/audit: baseline`

用于：Namespace

> 值**必须**是与 [Pod 安全标准](/zh/docs/concepts/security/pod-security-standards) 级别相对应的 `privileged`、`baseline` 或 `restricted` 之一。

Reviewer: ```suggestion
值**必须**是与 [Pod 安全标准](/zh/docs/concepts/security/pod-security-standards) 级别相对应的
`privileged`、`baseline` 或 `restricted` 之一。
```

> 具体来说，`audit` 标签不会阻止在带标签的 Namespace 中创建不符合指示级别概述要求的 Pod，但会向该 Pod 添加审计注解。

Reviewer: ```suggestion
具体来说，`audit` 标签不会阻止在带标签的 Namespace 中创建不符合指示级别要求的 Pod，
但会向该 Pod 添加审计注解。
```

> 请参阅 [在 Namespace 级别实施 Pod 安全性](/zh/docs/concepts/security/pod-security-admission)了解更多信息。

Reviewer: ```suggestion
请参阅[在名字空间级别实施 Pod 安全性](/zh/docs/concepts/security/pod-security-admission)了解更多信息。
```

<!--
### pod-security.kubernetes.io/audit-version

Example: `pod-security.kubernetes.io/audit-version: {{< skew latestVersion >}}`

Used on: Namespace

Value **must** be `latest` or a valid Kubernetes version in the format `v<MAJOR>.<MINOR>`.
This determines the version of the [Pod Security Standard](/docs/concepts/security/pod-security-standards) 
policies to apply when validating a submitted Pod.

See [Enforcing Pod Security at the Namespace Level](/docs/concepts/security/pod-security-admission)
for more information.
-->
### pod-security.kubernetes.io/audit-version {#pod-security-kubernetes-io-audit-version}

例子：`pod-security.kubernetes.io/audit-version: {{< skew latestVersion >}}`

用于：Namespace

值**必须**是 `latest` 或格式为 `v<MAJOR>.<MINOR>` 的有效 Kubernetes 版本。

> 这决定了在验证提交的 Pod 时要应用的  [Pod 安全标准](/zh/docs/concepts/security/pod-security-standards)策略的版本。

Reviewer: ```suggestion
此注解决定了在验证提交的 Pod 时要应用的  [Pod 安全标准](/zh/docs/concepts/security/pod-security-standards)策略的版本。
```

> 请参阅 [在 Namespace 级别实施 Pod 安全性](/zh/docs/concepts/security/pod-security-admission)了解更多信息。

Reviewer: ```suggestion
请参阅[在名字空间级别实施 Pod 安全性](/zh/docs/concepts/security/pod-security-admission)了解更多信息。
```

<!--
### pod-security.kubernetes.io/warn

Example: `pod-security.kubernetes.io/warn: baseline`

Used on: Namespace

Value **must** be one of `privileged`, `baseline`, or `restricted` which correspond to
[Pod Security Standard](/docs/concepts/security/pod-security-standards) levels. Specifically,
the `warn` label does not prevent the creation of a Pod in the labeled Namespace which does not meet the 
requirements outlined in the indicated level, but returns a warning to the user after doing so.
Note that warnings are also displayed when creating or updating objects that contain Pod templates,
such as Deployments, Jobs, StatefulSets, etc.

See [Enforcing Pod Security at the Namespace Level](/docs/concepts/security/pod-security-admission)
for more information.
-->
### pod-security.kubernetes.io/warn {#pod-security-kubernetes-io-warn}

例子：`pod-security.kubernetes.io/warn: baseline`

用于：Namespace

> 值**必须**是与 [Pod 安全标准](/zh/docs/concepts/security/pod-security-standards)级别相对应的 `privileged`、`baseline` 或 `restricted` 之一。具体来说，

Reviewer: ```suggestion
值**必须**是与 [Pod 安全标准](/zh/docs/concepts/security/pod-security-standards)级别相对应的
`privileged`、`baseline` 或 `restricted` 之一。特别地，
```

`warn` 标签不会阻止在带标签的 Namespace 中创建不符合指示级别概述要求的 Pod，但会在这样做后向用户返回警告。
请注意，在创建或更新包含 Pod 模板的对象时也会显示警告，例如 Deployment、Jobs、StatefulSets 等。

> 请参阅 [在 Namespace 级别实施 Pod 安全性](/zh/docs/concepts/security/pod-security-admission)了解更多信息。

Reviewer: ```suggestion
请参阅[在名字空间级别实施 Pod 安全性](/zh/docs/concepts/security/pod-security-admission)了解更多信息。
```

<!--
### pod-security.kubernetes.io/warn-version

Example: `pod-security.kubernetes.io/warn-version: {{< skew latestVersion >}}`

Used on: Namespace

Value **must** be `latest` or a valid Kubernetes version in the format `v<MAJOR>.<MINOR>`.
This determines the version of the [Pod Security Standard](/docs/concepts/security/pod-security-standards)
policies to apply when validating a submitted Pod. Note that warnings are also displayed when creating
or updating objects that contain Pod templates, such as Deployments, Jobs, StatefulSets, etc.

See [Enforcing Pod Security at the Namespace Level](/docs/concepts/security/pod-security-admission)
for more information.
-->
### pod-security.kubernetes.io/warn-version {#pod-security-kubernetes-io-warn-version}

例子：`pod-security.kubernetes.io/warn-version: {{< skew latestVersion >}}`

用于：Namespace

值**必须**是 `latest` 或格式为 `v<MAJOR>.<MINOR>` 的有效 Kubernetes 版本。

> 这决定了在验证提交的 Pod 时要应用的 [Pod 安全标准](/zh/docs/concepts/security/pod-security-standards)策略的版本。请注意，在创建或更新包含 Pod 模板的对象时也会显示警告，

Reviewer: ```suggestion
此注解决定了在验证提交的 Pod 时要应用的 [Pod 安全标准](/zh/docs/concepts/security/pod-security-standards)策略的版本。
请注意，在创建或更新包含 Pod 模板的对象时也会显示警告，
```

例如 Deployment、Jobs、StatefulSets 等。

> 请参阅 [在 Namespace 级别实施 Pod 安全性](/zh/docs/concepts/security/pod-security-admission)了解更多信息。

Reviewer: ```suggestion
请参阅[在名字空间级别实施 Pod 安全性](/zh/docs/concepts/security/pod-security-admission)了解更多信息。
```

<!--
### seccomp.security.alpha.kubernetes.io/pod (deprecated) {#seccomp-security-alpha-kubernetes-io-pod}

This annotation has been deprecated since Kubernetes v1.19 and will become non-functional in v1.25.
To specify security settings for a Pod, include the `securityContext` field in the Pod specification.
The [`securityContext`](/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context) field within a Pod's `.spec` defines pod-level security attributes.
When you [specify the security context for a Pod](/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod),
the settings you specify apply to all containers in that Pod.
-->
### seccomp.security.alpha.kubernetes.io/pod (已弃用) {#seccomp-security-alpha-kubernetes-io-pod}

此注解自 Kubernetes v1.19 起已被弃用，将在 v1.25 中失效。
要为 Pod 指定安全设置，请在 Pod 规范中包含 `securityContext` 字段。

> Pod 的 `.spec` 中的 [`securityContext`](/zh/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context) 字段定义了 pod 级别的安全属性。

Reviewer: ```suggestion
Pod 的 `.spec` 中的 [`securityContext`](/zh/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context)
字段定义了 Pod 级别的安全属性。
```

> 当你 [指定 Pod 的安全上下文](/zh/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod) 时，你指定的设置适用于该 Pod 中的所有容器。

Reviewer: ```suggestion
当你[为 Pod 设置安全上下文](/zh/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod) 时，
你所给出的设置适用于该 Pod 中的所有容器。
```

<!--
### container.seccomp.security.alpha.kubernetes.io/[NAME] {#container-seccomp-security-alpha-kubernetes-io}

This annotation has been deprecated since Kubernetes v1.19 and will become non-functional in v1.25.
The tutorial [Restrict a Container's Syscalls with seccomp](/docs/tutorials/clusters/seccomp/) takes
you through the steps you follow to apply a seccomp profile to a Pod or to one of
its containers. That tutorial covers the supported mechanism for configuring seccomp in Kubernetes,
based on setting `securityContext` within the Pod's `.spec`.
-->
### container.seccomp.security.alpha.kubernetes.io/[NAME] {#container-seccomp-security-alpha-kubernetes-io}

此注解自 Kubernetes v1.19 起已被弃用，将在 v1.25 中失效。

> 教程 [使用 seccomp 限制容器的系统调用](/zh/docs/tutorials/clusters/seccomp/) 将引导你完成将 seccomp 配置文件应用于 Pod 或其容器的步骤。

Reviewer: ```suggestion
教程[使用 seccomp 限制容器的系统调用](/zh/docs/tutorials/clusters/seccomp/)将引导你完成将
seccomp 配置文件应用于 Pod 或其容器的步骤。
```

该教程介绍了在 Kubernetes 中配置 seccomp 的支持机制，基于在 Pod 的 `.spec` 中设置 `securityContext`。

<!--
## Annotations used for audit

- [`pod-security.kubernetes.io/exempt`](/docs/reference/labels-annotations-taints/audit-annotations/#pod-security-kubernetes-io-exempt)
- [`pod-security.kubernetes.io/enforce-policy`](/docs/reference/labels-annotations-taints/audit-annotations/#pod-security-kubernetes-io-enforce-policy)
- [`pod-security.kubernetes.io/audit-violations`](/docs/reference/labels-annotations-taints/audit-annotations/#pod-security-kubernetes-io-audit-violations)

See more details on the [Audit Annotations](/docs/reference/labels-annotations-taints/audit-annotations/) page.
-->

> ## 用于审计的注解

Reviewer: ```suggestion
## 用于审计的注解    {#annonations-used-for-audit}
```

---

> title: 众所周知的标签、注解和污点

Reviewer 2: I would consider leaving “label”, “annotation” and “taint” untranslated.
Alternatively, make the English terms clear in the page introduction.

### failure-domain.beta.kubernetes.io/region (已弃用) {#failure-domainbetakubernetesioregion}

请参阅 [topology.kubernetes.io/region](#topologykubernetesioregion)。

{{< note >}} 从 v1.17 开始，此标签已弃用，取而代之的是 [topology.kubernetes.io/region](#topologykubernetesioregion)。 {{</note>}}

### failure-domain.beta.kubernetes.io/zone (已弃用) {#failure-domainbetakubernetesiozone}

请参阅 [topology.kubernetes.io/zone](#topologykubernetesiozone)。

{{< note >}} 从 v1.17 开始，此标签已弃用，取而代之的是 [topology.kubernetes.io/zone](#topologykubernetesiozone)。 {{</note>}}

### statefulset.kubernetes.io/pod-name {#statefulsetkubernetesiopod-name}

例子：`statefulset.kubernetes.io/pod-name=mystatefulset-7`

当 StatefulSet 控制器为 StatefulSet 创建 Pod 时，控制平面会在该 Pod 上设置此标签。标签的值是正在创建的 Pod 的名称。

有关详细信息，请参阅 StatefulSet 主题中的 [Pod 名称标签](/docs/concepts/workloads/controllers/statefulset/#pod-name-label)。

### topology.kubernetes.io/region {#topologykubernetesioregion}

例子：`topology.kubernetes.io/region=us-east-1`

请参阅 [topology.kubernetes.io/zone](#topologykubernetesiozone)。

### topology.kubernetes.io/zone {#topologykubernetesiozone}

例子：`topology.kubernetes.io/zone=us-east-1c`

> 用于：Node, PersistentVolume

Reviewer 2: ```suggestion
用于：Node、PersistentVolume
```

在 Node 上：`kubelet` 或外部`cloud-controller-manager` 使用 `cloudprovider` 提供的信息填充它。仅当你使用 `cloudprovider` 时才会设置此项。但是，如果它在你的拓扑中有意义，你应该考虑在 Node 上设置它。

> 在 PersistentVolume 上：拓扑感知卷配置器将自动在 `PersistentVolumes` 上设置 Node 亲和性约束。

Reviewer 2: I'd use singular terms for API kinds; for example:
```suggestion
在 PersistentVolume 上：拓扑感知卷配置器将自动在 `PersistentVolume` 上设置 Node 亲和性约束。
```

because English plurals aren't important to a document in Chinerse.

Author: Ok

一个 Zone 代表一个逻辑故障域。 Kubernetes 集群通常跨越多个 Zone 以提高可用性。虽然 Zone 的确切定义留给基础设施实现，但 Zone 的常见属性包括 Zone 内非常低的网络延迟、 Zone 内的免费网络流量以及与其他 Zone 的故障独立性。例如，一个 Zone 内的 Node 可能共享一个网络交换机，但不同 Zone 中的 Node 不应该。

一个 Region 代表一个更大的域，由一个或多个 Zone 组成。 Kubernetes 集群跨多个 Region 并不常见，虽然 Zone 或 Region 的确切定义留给基础设施实现，但 Region 的共同属性包括它们之间的网络延迟比它们内部更高，它们之间的网络流量成本非零，以及与其他 Zone 或 Region 的故障独立性。例如，一个 Region 内的 Node 可能共享电力基础设施（例如 UPS 或发电机），但不同 Region 的 Node 通常不会。

Kubernetes 对 Zone 和 Region 的结构做了一些假设：

> 1) Zone 和 Region 是分层的： Zone 是 Region 的严格子集，没有 Zone 可以在两个 Region 中
> 
> 2) Zone 名称跨 Region 是唯一的；例如， Region “africa-east-1”可能由 Zone “africa-east-1a”和“africa-east-1b”组成

Reviewer 2: ```suggestion
1. Zone 和 Region 是分层的： Zone 是 Region 的严格子集，没有 Zone 可以在两个 Region 中
1. Zone 名称跨 Region 是唯一的；例如， Region “africa-east-1”可能由 Zone “africa-east-1a”和“africa-east-1b”组成
```
?

如果 `PersistentVolumeLabel` 不支持你的 PersistentVolumes 的自动标签，你应该考虑手动添加标签（或添加对 `PersistentVolumeLabel` 的支持）。使用 `PersistentVolumeLabel` ，调度程序可以防止 Pod 将卷安装在不同的 Zone 中。如果你的基础架构没有此限制，则根本不需要将 Zone 标签添加到卷中。

### volume.beta.kubernetes.io/storage-provisioner (已弃用)

例子：`volume.beta.kubernetes.io/storage-provisioner: k8s.io/minikube-hostpath`

用于：PersistentVolumeClaim

此注解已被弃用。

### volume.kubernetes.io/storage-provisioner

用于：PersistentVolumeClaim

此注解将添加到动态配置所需的 PVC。

### node.kubernetes.io/windows-build {#nodekubernetesiowindows-build}

例子：`node.kubernetes.io/windows-build=10.0.17763`

用于：Node

当 kubelet 在 Microsoft Windows 上运行时，它会自动标记其 Node 以记录正在使用的 Windows Server 的版本。

标签的值采用“MajorVersion.MinorVersion.BuildNumber”格式。

### service.kubernetes.io/headless {#servicekubernetesioheadless}

例子：`service.kubernetes.io/headless=""`

用于：Service

当拥有的 Service 是无头类型时，控制平面将此标签添加到 Endpoints 对象。

### kubernetes.io/service-name {#kubernetesioservice-name}

例子：`kubernetes.io/service-name="nginx"`

用于：Service

Kubernetes 使用这个标签来区分多个服务。目前仅用于 `ELB` （弹性负载均衡器）。

### endpointslice.kubernetes.io/managed-by {#endpointslicekubernetesiomanaged-by}

例子：`endpointslice.kubernetes.io/managed-by="controller"`

> 用于：EndpointSlices

Reviewer 2: ```suggestion
用于：EndpointSlice
```
I'd use singular terms for API kinds

标签用于指示管理 EndpointSlice 的控制器或实体。该标签旨在使不同的 EndpointSlice 对象能够由同一集群内的不同控制器或实体管理。

### endpointslice.kubernetes.io/skip-mirror {#endpointslicekubernetesioskip-mirror}

例子：`endpointslice.kubernetes.io/skip-mirror="true"`

> 用于：Endpoints

Reviewer 2: BTW, this API kind is naturally plural (so “Endpoints” is correct).

Author: Endpoint maybe best?

Reviewer: No. What @sftim meant is that `Endpoints` is the resource name and it IS designed to be plural.

例子：`storageclass.kubernetes.io/is-default-class=true`

用于：StorageClass

当单个 StorageClass 资源将此注解设置为 `"true"` 时，未指定类的新 PersistentVolumeClaim 资源将被分配此默认类。

### alpha.kubernetes.io/provided-node-ip

例子：`alpha.kubernetes.io/provided-node-ip: "10.0.0.1"`

用于：Node

kubelet 可以在 Node 上设置此注解来表示其配置的 IPv4 地址。

当 kubelet 使用“外部”云提供商启动时，它会在 Node 上设置此注解以表示从命令行标志 ( `--node-ip` ) 设置的 IP 地址。云控制器管理器通过云提供商验证此 IP 是否有效。

### batch.kubernetes.io/job-completion-index

例子：`batch.kubernetes.io/job-completion-index: "3"`

用于：Pod

kube-controller-manager 中的 Job 控制器为使用 Indexed [完成模式](/docs/concepts/workloads/controllers/job/#completion-mode) 创建的 Pod 设置此注解。

### kubectl.kubernetes.io/default-container

例子：`kubectl.kubernetes.io/default-container: "front-end-app"`

注解的值是此 Pod 的默认容器名称。 例如，没有 `-c` 或 `--container` 标志的 `kubectl logs` 或 `kubectl exec` 将使用此默认容器。

### endpoints.kubernetes.io/over-capacity

例子：`endpoints.kubernetes.io/over-capacity:truncated`

用于：Endpoints

在 Kubernetes 集群 v1.22（或更高版本）中，如果 Endpoint 资源超过 1000 个， Endpoint 控制器会将此注解添加到 Endpoint 资源。注解表示 Endpoint 资源已超出容量，并且已将 Endpoint 数截断为 1000。

### batch.kubernetes.io/job-tracking

例子：`batch.kubernetes.io/job-tracking: ""`

> 用于：Jobs

Reviewer 2: ```suggestion
用于：Job
```

作业上存在此注解表明控制平面正在[使用 Finalizer 追踪 Job](/docs/concepts/workloads/controllers/job/#job-tracking-with-finalizers)。
你应该**不**手动添加或删除此注解。

### scheduler.alpha.kubernetes.io/preferAvoidPods (deprecated) {#scheduleralphakubernetesio-preferavoidpods}

> 用于：Nodes

Reviewer 2: ```suggestion
用于：Node
```

-->
### kubernetes.io/change-cause {#change-cause}

例子：`kubernetes.io/change-cause=kubectl edit --record deployment foo`

用于：所有对象

此注解是对更改某些内容的最佳猜测。

将 `--record` 添加到可能会更改对象的 `kubectl` 命令时会填充它。

### kubernetes.io/description {#description}

例子：`kubernetes.io/description: "Description of K8s object."`

用于：所有对象

此注解用于描述给定对象的特定行为。

### kubernetes.io/enforce-mountable-secrets {#enforce-mountable-secrets}

例子：`kubernetes.io/enforce-mountable-secrets: "true"`

用于：ServiceAccount

此注解的值必须为 **true** 才能生效。此注解表示作为此服务帐户运行的 pod 只能引用在服务帐户的 `secrets` 字段中指定的 Secret API 对象。

<!--
### controller.kubernetes.io/pod-deletion-cost {#pod-deletion-cost}

Example: `controller.kubernetes.io/pod-deletion-cost=10`

Used on: Pod

This annotation is used to set [Pod Deletion Cost](/docs/concepts/workloads/controllers/replicaset/#pod-deletion-cost)
which allows users to influence ReplicaSet downscaling order. The annotation parses into an `int32` type.

### beta.kubernetes.io/instance-type (deprecated)
-->

### controller.kubernetes.io/pod-deletion-cost {#pod-deletion-cost}

例子：`controller.kubernetes.io/pod-deletion-cost=10`

用于：Pod

> 该注解用于设置 [Pod 删除成本](/docs/concepts/workloads/controllers/replicaset/#pod-deletion-cost) 允许用户影响 ReplicaSet 缩减顺序。 注解解析为 `int32` 类型。

Reviewer: ```suggestion
该注解用于设置 [Pod 删除成本](/zh/docs/concepts/workloads/controllers/replicaset/#pod-deletion-cost)，
允许用户影响 ReplicaSet 缩减顺序。注解值解析为 `int32` 类型。
```

### beta.kubernetes.io/instance-type (已弃用) {#beta-kubernetes-io-instance-type}

<!--
Starting in v1.17, this label is deprecated in favor of [node.kubernetes.io/instance-type](#nodekubernetesioinstance-type).
-->
{{< note >}} 从 v1.17 开始，此标签已弃用，取而代之的是 [node.kubernetes.io/instance-type](#nodekubernetesioinstance-type)。 {{< /note >}}

<!--
### node.kubernetes.io/instance-type {#nodekubernetesioinstance-type}

Example: `node.kubernetes.io/instance-type=m3.medium`

Used on: Node

The Kubelet populates this with the instance type as defined by the `cloudprovider`.
This will be set only if you are using a `cloudprovider`. This setting is handy
if you want to target certain workloads to certain instance types, but typically you want
to rely on the Kubernetes scheduler to perform resource-based scheduling. You should aim to schedule based on properties rather than on instance types (for example: require a GPU, instead of requiring a `g2.2xlarge`).
-->
### node.kubernetes.io/instance-type {#nodekubernetesioinstance-type}

例子：`node.kubernetes.io/instance-type=m3.medium`

用于：Node

> Kubelet 使用 `cloudprovider` 定义的实例类型填充它。

Reviewer: ```suggestion
Kubelet 使用 `cloudprovider` 所定义的实例类型填充它。
```

> 仅当你使用 `cloudprovider` 时才会设置此项。如果你希望将某些工作负载定位到某些实例类型，则此设置非常方便，但通常你希望依靠 Kubernetes 调度程序来执行基于资源的调度。

Reviewer: ```suggestion
仅当你使用 `cloudprovider` 时才会设置此项。如果你希望将某些工作负载运行到某些实例类型上，
此设置非常方便，但通常你希望依靠 Kubernetes 调度程序来执行基于资源的调度。
```

See [topology.kubernetes.io/zone](#topologykubernetesiozone).
-->
### failure-domain.beta.kubernetes.io/zone (已弃用) {#failure-domainbetakubernetesiozone}

请参阅 [topology.kubernetes.io/zone](#topologykubernetesiozone)。

<!--
Starting in v1.17, this label is deprecated in favor of [topology.kubernetes.io/zone](#topologykubernetesiozone). 
-->
{{< note >}} 从 v1.17 开始，此标签已弃用，取而代之的是 [topology.kubernetes.io/zone](#topologykubernetesiozone)。 {{</note>}}

<!--
### statefulset.kubernetes.io/pod-name {#statefulsetkubernetesiopod-name}

Example:

`statefulset.kubernetes.io/pod-name=mystatefulset-7`

When a StatefulSet controller creates a Pod for the StatefulSet, the control plane
sets this label on that Pod. The value of the label is the name of the Pod being created.

See [Pod Name Label](/docs/concepts/workloads/controllers/statefulset/#pod-name-label) in the
StatefulSet topic for more details.
-->
### statefulset.kubernetes.io/pod-name {#statefulsetkubernetesiopod-name}

例子：`statefulset.kubernetes.io/pod-name=mystatefulset-7`

当 StatefulSet 控制器为 StatefulSet 创建 Pod 时，控制平面会在该 Pod 上设置此标签。标签的值是正在创建的 Pod 的名称。

> 有关详细信息，请参阅 StatefulSet 主题中的 [Pod 名称标签](/docs/concepts/workloads/controllers/statefulset/#pod-name-label)。

Reviewer: ```suggestion
有关详细信息，请参阅 StatefulSet 主题中的 [Pod 名称标签](/zh/docs/concepts/workloads/controllers/statefulset/#pod-name-label)。
```

-->
### topology.kubernetes.io/zone {#topologykubernetesiozone}

例子：`topology.kubernetes.io/zone=us-east-1c`

用于：Node、PersistentVolume

在 Node 上：`kubelet` 或外部`cloud-controller-manager` 使用 `cloudprovider` 提供的信息填充它。仅当你使用 `cloudprovider` 时才会设置此项。
但是，如果它在你的拓扑中有意义，你应该考虑在 Node 上设置它。

在 PersistentVolume 上：拓扑感知卷配置器将自动在 `PersistentVolume` 上设置 Node 亲和性约束。

一个 Zone 代表一个逻辑故障域。 Kubernetes 集群通常跨越多个 Zone 以提高可用性。虽然 Zone 的确切定义留给基础设施实现，
但 Zone 的常见属性包括 Zone 内非常低的网络延迟、 Zone 内的免费网络流量以及与其他 Zone 的故障独立性。
例如，一个 Zone 内的 Node 可能共享一个网络交换机，但不同 Zone 中的 Node 不应该。

一个 Region 代表一个更大的域，由一个或多个 Zone 组成。Kubernetes 集群跨多个 Region 并不常见，虽然 Zone 或 Region 的确切定义留给基础设施实现，
但 Region 的共同属性包括它们之间的网络延迟比它们内部更高，它们之间的网络流量成本非零，以及与其他 Zone 或 Region 的故障独立性。
例如，一个 Region 内的 Node 可能共享电力基础设施（例如 UPS 或发电机），但不同 Region 的 Node 通常不会。

Kubernetes 对 Zone 和 Region 的结构做了一些假设：

1. Zone 和 Region 是分层的： Zone 是 Region 的严格子集，没有 Zone 可以在两个 Region 中

2. Zone 名称跨 Region 是唯一的；例如， Region “africa-east-1”可能由 Zone “africa-east-1a”和“africa-east-1b”组成

> 假设拓扑标签不会改变应该是安全的。尽管标签是严格可变的，但它们的消费者可以假设给定 Node 不会在 Zone 之间移动而不会被销毁和重新创建。

Author: 好的
